x/auth/ante.go

```diff
@@ -77,7 +77,7 @@ func NewAnteHandler(am sdk.AccountMapper, feeHandler sdk.FeeHandler) sdk.AnteHan
 				// TODO: min fee
 				if !fee.Amount.IsZero() {
 					signerAcc, res = deductFees(signerAcc, fee)
-					feeHandler(ctx, tx, fee.Amount)
+					feeHandler(ctx, fee.Amount)
```

Comment:
I disagree, the transaction should not explicitly be handled by the feeHandler, the transaction should be handled by the appropriate handler which the transaction is destined for - that handler can then modify the context, or update the keeper which wraps the fee handler (such as in x/fee_distribution) which is processing the fees to modify the way in which fees are processed. Remember the tx here is not specific to the feeHandler it's an arbitrary std transaction specific to another module, I'd rather that other module process specifics to fees and inform the fee handler outside than have to design an specific fee handler which is aware a certain transaction types

Reply:
I agree that we don't want to handle any *transaction* logic here, but applications might want a way for fee processing to depend on transaction data, which the handlers can't do easily since they just have `Msg`s, not fee information. Seems like "informing the fee handler" might lead to some roundabout logic.

I don't have a particular usecase in mind here though, so we can leave as-is for now.

Reply:
cool

Reply:
I would suspect we'd want the tx to be passed in here too, but its not straight forward because of type assertions ... We could be more clear about what works with Tx and what works with StdTx